tests/unit/strict-resolver/basic-test.js

```diff
@@ -104,34 +106,131 @@ test("can lookup something in another namespace with different syntax", function
   assert.equal(adapter, expected, 'default export was returned');
 });
 
-test("can lookup a view in another namespace", function(assert) {
+test("can lookup a helper", function(assert) {
   assert.expect(3);
 
-  let expected = { isViewFactory: true };
-  define('other/views/post', [], function(){
-    assert.ok(true, "view was invoked properly");
+  let expected = { isHelperInstance: true };
+  define('foo-bar/helpers/reverse-list', [], function(){
+    assert.ok(true, "helper was invoked properly");
 
     return { default: expected };
   });
 
-  var view = resolver.resolve('other@view:post');
+  var helper = resolver.resolve('helper:reverse-list');
 
-  assert.ok(view, 'view was returned');
-  assert.equal(view, expected, 'default export was returned');
+  assert.ok(helper, 'helper was returned');
+  assert.equal(helper, expected, 'default export was returned');
 });
 
-test("can lookup a view in another namespace with different syntax", function(assert) {
+test('can lookup an engine', function(assert) {
   assert.expect(3);
 
-  let expected = { isViewFactory: true };
-  define('other/views/post', [], function(){
-    assert.ok(true, "view was invoked properly");
+  let expected = {};
+  define('foo-bar/engine', [], function(){
+    assert.ok(true, 'engine was invoked properly');
+
+    return { default: expected };
+  });
+
+  let engine = resolver.resolve('engine:foo-bar');
+
+  assert.ok(engine, 'engine was returned');
+  assert.equal(engine, expected, 'default export was returned');
+});
+
+test('can lookup a route-map', function(assert) {
+  assert.expect(3);
+
+  let expected = { isRouteMap: true };
+  define('foo-bar/routes', [], function(){
+    assert.ok(true, 'route-map was invoked properly');
+
+    return { default: expected };
+  });
+
+  let routeMap = resolver.resolve('route-map:foo-bar');
+
+  assert.ok(routeMap, 'route-map was returned');
+  assert.equal(routeMap, expected, 'default export was returned');
+});
+
+test('warns if looking up a camelCase name', function(assert){
+  assert.expect(2);
+
+  define('foo-bar/helpers/reverse-list', [], function(){
+    return { default: { isHelperInstance: true } };
+  });
+
+  Ember.assert = function(message, test) {
+    if (!test) {
+      assert.equal(message, 'Attempted to lookup "helper:reverseList". Use "helper:reverse-list" instead.');
+    }
+  };
+
+  var helper = resolver.resolve('helper:reverseList');
+  assert.ok(!helper, 'no helper was returned');
+});
+
+skip('errors if lookup of a route-map does not specify isRouteMap', function(assert) {
```

Comment:
Oh, I rewote the description and left this out. So the skipped tests are basically questions about wether that is behavior we want to support... was looking for your feedback on that...